tests/tests/test-txpool/test-txpool-fairness.ts

```diff
@@ -0,0 +1,273 @@
+    expect(transferExts[1].tip.toBigInt()).to.eq(MED_TIP);
+    expect(transferExts[2].tip.toBigInt()).to.eq(NO_TIP);
+  });
+
+  it("should treat eth and substrate txns fairly", async function () {
+    context.ethTransactionType = "EIP1559";
+
+    // tip 1 and 3 will be substrate txns, we express their tip above as per-gas but must send them
+    // expressed as a flat tip. So we query the weight and convert to gas, then multiply by the
+    // per-gas tip. We do this because it's the inverse of the txpool algo, and we want to show that
+    // this algo will order txns by tip in this test.
+    //
+    // so the expected order is:
+    // tip_0 (eth)
+    // tip_1 (substrate)
+    // tip_2 (eth)
+    // tip_3 (substrate)
+    const TIP_PER_GAS_0 = 10000n;
+    const TIP_PER_GAS_1 = 20000n;
+    const TIP_PER_GAS_2 = 30000n;
+    const TIP_PER_GAS_3 = 40000n;
+
+    // here we query the weight of a substrate balance transfer
+    const dummyTransfer = context.polkadotApi.tx.balances.transfer(alith.address, GLMR);
+    const info = await context.polkadotApi.rpc.payment.queryInfo(dummyTransfer.toHex());
+    const weight = info.weight.toBigInt();
+    const balances_transfer_effective_gas = weight / WEIGHT_PER_GAS;
+
+    // tx0 is an eth txn
+    const tx0 = await createTransfer(context, ethan.address, 1, {
+      from: charleth.address,
+      privateKey: CHARLETH_PRIVATE_KEY,
+      maxFeePerGas: HIGH_MAX_FEE_PER_GAS,
+      maxPriorityFeePerGas: "0x" + TIP_PER_GAS_0.toString(16),
+    });
+
+    // tx1 is a substrate txn
+    const tx1 = await context.polkadotApi.tx.balances
+      .transfer(ethan.address, GLMR)
+      .signAsync(alith, { tip: TIP_PER_GAS_1 * balances_transfer_effective_gas });
+
+    // tx2 is an eth txn
+    const tx2 = await createTransfer(context, ethan.address, 1, {
+      from: dorothy.address,
+      privateKey: DOROTHY_PRIVATE_KEY,
+      maxFeePerGas: HIGH_MAX_FEE_PER_GAS,
+      maxPriorityFeePerGas: "0x" + TIP_PER_GAS_2.toString(16),
+    });
+
+    // tx3 is a substrate txn
+    const tx3 = await context.polkadotApi.tx.balances
+      .transfer(ethan.address, GLMR)
+      .signAsync(baltathar, { tip: TIP_PER_GAS_3 * balances_transfer_effective_gas });
+
+    const result = await context.createBlock([
+      // use an order other than by priority
+      tx2,
+      tx3,
+      tx0,
+      tx1,
+    ]);
+
+    // get and filter the block's extrinsics
+    const hash = result.block.hash;
+    const apiAt = await context.polkadotApi.at(hash);
+    const { block } = await context.polkadotApi.rpc.chain.getBlock(hash);
+    let transferExts = block.extrinsics.filter((ext) => {
+      return (
+        (ext.method.section == "balances" && ext.method.method == "transfer") ||
+        (ext.method.section == "ethereum" && ext.method.method == "transact")
+      );
+    });
+
+    expect(transferExts.length).to.eq(4);
+    expect(transferExts[0].method.section).to.eq("balances");
```

Comment:
We should also assert that the correct TX type was ordered in accordance with the Tip and not just the type (since there are 2 of each we can't be sure that the appropriate tx was picked first)